### PR TITLE
Issue 5759: Dev Guide - 1st Code Walk through

### LINKED ISSUE #5759: Documentation : Dev Guide - 1st Code Walk through
**Problem description**
Simplify the existing Dev Guide to create applications against Pravega.

**Problem location**
Documentation.

**Suggestions for an improvement**
Improve it.

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
Proposed Dev Guide - 1st Code walkthrough

**Purpose of the change**  
Fixes #5759 

**What the code does**  
Dev Guide - 1st Code walkthrough for a sample Pravega application.

**How to verify it**  
Documentation change.

### CODE REVIEW

File: documentation/src/docs/beginner_dev_guide.md
Comment (SrishT) [inline documentation/src/docs/beginner_dev_guide.md:80]: The hyperlink for the text `GitHub Releases` is not showing up correctly

**Review by SrishT — APPROVED** [review]
LGTM. I was able to walk through the guide step-by-step and create a small sample app successfully.

File: documentation/src/docs/beginner_dev_guide.md
Comment (shrids) [reply documentation/src/docs/beginner_dev_guide.md:81]: fixed.

**Comment by shrids** [comment]
@derekm @RaulGracia I have addressed your review comments as part of PR: https://github.com/pravega/pravega/pull/5745

File: documentation/src/docs/beginner_dev_guide.md
Comment (RaulGracia) [inline documentation/src/docs/beginner_dev_guide.md:14]: Perhaps, with mkdocs items like these require a blank line in between.

File: documentation/src/docs/beginner_dev_guide.md
Comment (RaulGracia) [inline documentation/src/docs/beginner_dev_guide.md:23]: Same here, perhaps a blank line is needed.

File: documentation/src/docs/beginner_dev_guide.md
Comment (RaulGracia) [inline documentation/src/docs/beginner_dev_guide.md:132]: `Let’s`, `We’ll` -> looks like the `'` character is not right. 
In the paragraph below, it looks right in `Pravega let's`.

**Review by RaulGracia — APPROVED** [review]
Looks good, just couple of minor points.

File: documentation/src/docs/beginner_dev_guide.md
Comment (shrids) [reply documentation/src/docs/beginner_dev_guide.md:14]: added it.

File: documentation/src/docs/beginner_dev_guide.md
Comment (shrids) [reply documentation/src/docs/beginner_dev_guide.md:23]: added it

File: documentation/src/docs/beginner_dev_guide.md
Comment (shrids) [reply documentation/src/docs/beginner_dev_guide.md]: fixed it :)

**Comment by derekm** [comment]
Latest changes are visible in the nightly docs:

https://pravega.io/docs/nightly/beginner_dev_guide/

File: documentation/src/docs/beginner_dev_guide.md
Comment (RaulGracia) [inline documentation/src/docs/beginner_dev_guide.md:2]: The copyright header has changed, please fix it.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Holding this one until #5882 is merged. Then, we can proceed adding this content in the new structure of docs.

File: documentation/src/docs/beginner_dev_guide.md
Comment (shrids) [reply documentation/src/docs/beginner_dev_guide.md]: Thanks, fixed it.

**Comment by RaulGracia** [comment]
@shrids can you add an entry for this document as a first element in the `Developing Applications` section of the documentation nav panel?

**Review by RaulGracia — APPROVED** [review]

**Comment by shrids** [comment]
@RaulGracia I have updated mkdocs.yml with this change.

**Pull request merged**
